### ISSUE TITLE
New feature request : Fm transmission

### ISSUE THREAD

**Issue by iztoogood** [openDsh/dash#13]
I have a new requet ... a feature addition ...
May you add https://github.com/miegl/PiFmAdv
it's a fork off piFm ... but this one allow to choose the gpio for the antena wire (usefull while using small tft screen) ... and it can broadcast messages over rds

I would allow to send all openDsh/dash ia audio to the original car radio

**Comment by WardBenjamin** [comment]
I'm not sure what the current CPU usage of the project is while running OpenAuto, but PiFmAdv cites CPU usage of 33-40% solely for audio transmission (according to the README), so that could be problematic.

**Comment by rsjudka** [comment]
Yeah we actually talked about this a bit on Slack. Two of the biggest concerns are the performance and whether the Pi can keep up, and then the "legality" of actually implementing this into dash.

I'll still leave this issue open in case things change in the future, but for now I don't think this is a feasible feature to add

**Comment by iztoogood** [comment]
Ok, it was just a suggestion, for my part i'm trying to out all system sound to PiFmAdv 
i'm trying to adapt the steps discrip here https://www.reddit.com/r/raspberry_pi/comments/7igu4u/redirect_system_audio_output_to_pifm/
to PiFmAdv instead of piFm (or piFmRds)
the Adv fork allow to choose the gpio pin 

thanks for your feedbacks

**Comment by davisonja** [comment]
> I have a new requet ... a feature addition ...
> May you add https://github.com/miegl/PiFmAdv
> it's a fork off piFm ... but this one allow to choose the gpio for the antena wire (usefull while using small tft screen) ... and it can broadcast messages over rds
> 
> I would allow to send all openDsh/dash ia audio to the original car radio

I was never entirely sure I was comfortable with my pi doing this in the end. Various discussions I've seen (which to be fair, when I was looking was a good many years ago now) had scary stories of side effects with frequency echos (IIRC). I decided I'd be better off with an external already compliant transmitter. At the time that turned into the hardware linked to https://github.com/Manawyrm/FMBerry, which I acquired a few of. There are more modern alternatives, such as the  Si4713-B30 (which I found at https://www.mikroe.com/radiostation-click in an actual board).

I've been using the FMBerry code to run the audio out of my Pi4 (analogue 'headphone' socket) with the old 7" 'official' touchscreen. Stock Raspbian with android mobile bluetooth-audio. Which means phone -> pi -> car stereo. It's the interim solution before finding a decent 'headunit' base to use, which this looks like it is! Pity the slack invites keep expiring ;)

It has the disadvantage that it's not built onto the pi, but it's a small module, does RDS and works quite effectively powered entirely off the Pi 3v3 lines.

Now all I need is to build an interface to run in openDsh so I can change the frequency. We found a few places with a broadcast station on the same frequency, which we got stuck listening to until the pi turned on enough to start the FMBerry controller, and caused a few audio glitches in very specific spots.

**Comment by rsjudka** [comment]
here's the (non-expiring) slack link! https://join.slack.com/t/opendsh/shared_invite/zt-la398uly-a6eMH5ttEQhbtE6asVKx4Q

also added it to the repo "about" section :)

not sure how familiar you are with openDsh, but there is an external app launcher that would work great with that (assuming it plays nicely with xlib). Definitely something we could also consider building directly into dash, just would need to familiarize myself more with how all that works lol

**Comment by davisonja** [comment]
Thank you :)

I've only just found openDsh, tbh. It ticks all the things I wanted, tho, in that it's open, extendable and best of all it just works on top of a standard Raspbian on the 7" touchscreen but 'inverted' - including android auto. In addition to the FM transmitter i'm using I've also got flakey fuel senders in my Nissan, so the dashboard gauge almost always shows empty, but I've got an app (and bluetooth ODBII adaptor) which shows useful info, which I'm wanting to integrate into the headunit as well.

So. Not too familiar, but will be getting more familiar as I build what I need. More than happy to PR things in, too. :)

**Comment by memtech3** [comment]
Although it is unlikely, there is a chance that a crude FM transmitter could cause interference to mission critical and or life safety communications solutions (public safety and business two way radio systems, SCADA systems, alarm systems, aviation communication systems, drone controllers, and other miscellaneous radio communications solutions). Additionally, the FCC has been ramping up enforcement action against pirate radio station operators. For liability purposes, I would recommend that the opendsh project stay away from the idea of pirate radio solutions unless y'all can get a professional, documented, legal opinion from an attorney or the FCC. Let someone else develop a plugin that adds pirate radio functionality independent of the opendsh organization.

It's an interesting feature suggestion but I think it just brings too much liability to this project and I'd rather not see it sued into oblivion (not crazy likely but possible in litigious America)  before I get a chance to install it :wink: 

**DISCLAIMER: I AM NOT AN ATTORNEY AND MY ADVICE COMES FROM MY SOMETIMES PEA SIZED BRAIN. I MEAN WELL TO ALL IN THIS DISCUSSION**

**Comment by davisonja** [comment]
I would imagine using an existing FM transmitter that had (things like) FCC
approval would have no issues.

Certainly in New Zealand there are a variety of devices that rebroadcast
audio from Bluetooth to FM frequencies, specifically for listening to
aftrrmarket audio on stock car radios - they escape major issues by having
a very limited range (in the order of a metre or two) aside from any other
compliance measures. The fm transmitter i use is one that originally was
distributed/sold by Sony as an attachment for one of their cell phones.

There's a fun project somewhere that generates fm direct from a pi which
certainly had a number of people concerned with the impact it might have on
nearby systems, but I don't know that anyone is suggesting using that.

**Comment by memtech3** [comment]
@davisonja I believe the issue suggested adding support for pi gpio fm transmission

**Comment by davisonja** [comment]
@memtech3 sorry, you're entirely correct on that, I should have checked the issue before replying (which I did via email); I agree it's a bad plan to genereate FM from the pi itself. Solid support for FM transmission is a very good idea, in contrast.
